src/universal/modules/meeting/ducks/meeting.js

```diff
@@ -0,0 +1,166 @@
+import {Map as iMap, List as iList} from 'immutable';
+import {push, replace} from 'react-router-redux';
+import {fetchGraphQL} from '../../../utils/fetching';
+import {ensureState} from 'redux-optimistic-ui';
+import {localStorageVars} from '../../../utils/clientOptions';
+import socketCluster from 'socketcluster-client';
+
+// Changed string consts because chances are we'll have more than 1 "CREATE_SUCCESS" down the road
+export const CREATE_MEETING_REQUEST = 'action/meeting/CREATE_MEETING_REQUEST';
+export const CREATE_MEETING_ERROR = 'action/meeting/CREATE_MEETING_ERROR';
+export const CREATE_MEETING_SUCCESS = 'action/meeting/CREATE_MEETING_SUCCESS';
+const SET_MEETING_ID = 'action/meeting/SET_MEETING_ID';
+// TODO is a request necessary?
+export const UPDATE_MEETING_REQUEST = 'action/meeting/UPDATE_MEETING_REQUEST';
+export const UPDATE_MEETING_ERROR = 'action/meeting/UPDATE_MEETING_ERROR';
+export const UPDATE_MEETING_SUCCESS = 'action/meeting/UPDATE_MEETING_SUCCESS';
+//TODO multiple meetings at once? It's possible with redux-operations
+// making the switch to redux-operations now is cheap, later on it'll become a pain to switch
+
+const initialState = iMap({
+  isLoading: false,
+  isLoaded: false,
+  instance: iMap({
+    id: '',
+    content: '',
+    lastUpdatedBy: '',
+    currentEditors: iList()
+  }),
+  mySocketId: '',
+  otherEditing: false
+});
+
+export default function reducer(state = initialState, action = {}) {
+  switch (action.type) {
+    case SET_MEETING_ID:
+      return state.mergeDeep({
+        instance: iMap({
+          id: action.payload.id
+        })
+      })
+    case CREATE_MEETING_REQUEST:
+      return state.merge({
+        isLoading: true
+      });
+    case CREATE_MEETING_SUCCESS:
+      return state.mergeDeep({
+        isLoading: false,
+        isLoaded: true,
+        instance: iMap({
+          id: action.payload.id
+        })
+      });
+    case CREATE_MEETING_ERROR:
+    case UPDATE_MEETING_ERROR:
+      return state.merge({
+        isLoading: false,
+        error: action.error
+      });
+    case UPDATE_MEETING_SUCCESS:
+      return state.mergeDeep({
+        instance: iMap(action.payload)
+      });
+    default:
+      return state
+  }
+};
+
+export const setMeetingId = meetingId => {
+  return {
+    type: SET_MEETING_ID,
+    payload: {id: meetingId}
+  }
+}
+
+export const createMeetingAndRedirect = () => {
+  return async function (dispatch, getState) {
+    const query = `
+    mutation {
+      payload: createMeeting {
+        id
+      }
+    }`;
+    const {error, data} = await fetchGraphQL({query});
```

Comment:
Ugh. So much more transparent to first-time readers. No implicit redux middleware. Much better!